featurebyte/models/request_input.py

```diff
@@ -148,20 +151,50 @@ async def materialize(
         """
         query_expr = self.get_query_expr(source_info=session.get_source_info())
 
+        # estimate the maximum POINT_IN_TIME value to filter the sample
+        column_names_and_dtypes = await self.get_column_names_and_dtypes(session=session)
+        available_columns = list(column_names_and_dtypes.keys())
+        if self.columns is None:
+            columns = available_columns
+        else:
+            columns = self.columns
+
+        if self.columns_rename_mapping is not None:
+            output_columns = [self.columns_rename_mapping.get(col, col) for col in columns]
+            column_names_and_dtypes = {
+                self.columns_rename_mapping.get(col, col): dtype
+                for col, dtype in column_names_and_dtypes.items()
+            }
```

Comment:
nit: Do we need to consider the possibility of column name conflict here (user renames the column to existing column name which causes more than 1 keys with the same name).
            }

Reply:
This is an existing issue. I think we could update https://github.com/featurebyte/featurebyte/blob/23f3921586941cd3234dc08e698463b91ddad979/featurebyte/models/request_input.py#L214-L223 to prevent column name conflict.